### PR TITLE
♻️(frontend) split factories file into two

### PR DESCRIPTION
**Pull request by rlecellier**
## Purpose

Joanie and Richie apis share some model names.
We are in need of a proper separation to avoid
naming nightmare.

[Linked PR](https://github.com/openfun/richie/pull/1940) (shared CourseGlimpse between Joanie and Richie will introduce two Course types and factories).

### CODE REVIEW

File: UPGRADE.md [hunk 1]
```diff
@@ -18,6 +18,8 @@ $ make migrate
 
 ## 2.20.1 to 2.21.0
 
+- utils/test/factories.ts have been split into multiple files: joanie.ts, richie.ts and 
+  reactQuery.ts. Update your import accordingly.
```
Comment (jbpenrath) [inline UPGRADE.md:22]: Should be moved up into the `Unreleased` section

File: src/frontend/js/utils/test/factories/richie.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+});
+
+export const FonzieUserFactory = compose(
+  UserFactory,
+  createSpec({
+    access_token: btoa(faker.datatype.uuid()),
+  }),
+);
+
+export const RichieContextFactory = (context: Partial<CommonDataProps['context']> = {}) =>
```
Comment (jbpenrath) [inline src/frontend/js/utils/test/factories/richie.ts:43]: Why not conserve `ContextFactory` and prevent too much changes ? Furthermore, there is no duplication issue about this factory.
```suggestion
export const ContextFactory = (context: Partial<CommonDataProps['context']> = {}) =>
```

File: src/frontend/js/utils/test/factories/reactQuery.ts
Comment (jbpenrath) [inline src/frontend/js/utils/test/factories/reactQuery.ts:1]: 🏅

File: src/frontend/js/utils/test/factories/richie.ts (hunk 2 above)
Comment (rlecellier) [reply src/frontend/js/utils/test/factories/richie.ts:43]: `Context` is kind of a reserved word in React.
here we talk about the richie's application context and not a ReactContextFactory.

Comment (jbpenrath) [reply src/frontend/js/utils/test/factories/richie.ts:43]: Mokay

**Review by jbpenrath — APPROVED** [review]
LGTM ✅

**Pull request merged**